From d6f2c593fcdfa7d0492e23ef34b32320652ba453 Mon Sep 17 00:00:00 2001
From: Kent Tamura <tkent@chromium.org>
Date: Thu, 16 May 2024 08:08:49 +0000
Subject: [PATCH] RubyLB: Fix a crash in InlineItemResult::ToString()

GetLayoutObject() can be null for an <rt> without a <ruby>.

Bug: 324111880
Change-Id: I8bbece86b9470ed04b4bb3d494cc6212e8de6b54
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5539141
Commit-Queue: Koji Ishii <kojii@chromium.org>
Auto-Submit: Kent Tamura <tkent@chromium.org>
Reviewed-by: Koji Ishii <kojii@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1301805}
---

--- a/third_party/blink/renderer/core/layout/inline/inline_item_result.cc
+++ b/third_party/blink/renderer/core/layout/inline/inline_item_result.cc
@@ -81,7 +81,11 @@
         ifc_text_content.Substring(TextOffset().start, TextOffset().Length())
             .EncodeForDebugging());
   } else if (IsRubyColumn()) {
-    builder.Append(item->GetLayoutObject()->ToString());
+    if (item->GetLayoutObject()) {
+      builder.Append(item->GetLayoutObject()->ToString());
+    } else {
+      builder.Append("(anonymous)");
+    }
     builder.Append(", base_line: [\n");
     String child_indent = indent + "\t";
     for (const auto& r : ruby_column->base_line.Results()) {
